### PR TITLE
Completed the assignment addition form.

### PR DESCRIPTION
**Pull request by m-e-l-u-h-a-n**
- Implemented the requested changes.
- Date input component added for better ux.
- Added better interface for adding review criterions.
- Fixed the bug in editing a criterion.
**Note:**
Bug while editing a task exists even now due shallow copy of object issue, will be fixed during api integration..

### CODE REVIEW

**Comment by king-11** [comment]
I guess eslint is causing issue again :(

File: components/DateInput.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    transition="scale-transition"
+    offset-y
+    min-width="250px"
+    @update:return-value="emitDate"
+  >
+    <template v-slot:activator="{on, attrs}">
+      <v-text-field
+        v-model="date"
+        color="primary"
+        label="Submission deadline"
```
Comment (king-11) [inline components/DateInput.vue:16]: take it as prop instead

File: pages/createassignment.vue [hunk 2]
```diff
@@ -0,0 +1,304 @@
+/* eslint-disable no-console */ /* eslint-disable
+vue/no-side-effects-in-computed-properties */
+<template>
+  <v-main>
+    <v-row row wrap justify="center" align="center" class="mt-n5">
+      <v-col cols="12" md="10" xl="8">
```
Comment (king-11) [inline pages/createassignment.vue:6]: change to v-container as v-main is used only for layouts and also add a bit of margin or padding on top

File: pages/createassignment.vue [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+                counter
+                maxlength="250"
+                :rules="[rules.descriptionLength]"
+                :outlined="breakpoint"
+              />
+              <v-row>
+                <v-col sm="6">
+                  <date-input @update:date="(updatedDate) => {submissionDeadline = updatedDate}" />
+                </v-col>
+                <v-col sm="6">
```
Comment (king-11) [inline pages/createassignment.vue:35]: pass in the prop for name

File: components/AddTask.vue [hunk 4]
```diff
@@ -0,0 +1,284 @@
+/* eslint-disable no-console */
+<template>
+  <div>
+    <v-layout child-flex>
+      <v-card flat>
+        <v-card-title class="info--text">
+          {{ getTitle }}
```
Comment (king-11) [inline components/AddTask.vue:7]: not needed i guess

File: components/AddTask.vue [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+        </v-card-title>
```
Comment (king-11) [inline components/AddTask.vue:8]: add a max-width otherwise looks odd

File: components/ReviewPallete.vue [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+      type: Object,
+      default () {
+        return {
+          ques: 'Untitled Criterion',
+          points: 0
+        }
+      }
+    },
+    // because user may enter a NaN and that would produse unwanted vue-warn in console.
+    // eslint-disable-next-line vue/require-prop-types
```
Comment (king-11) [inline components/ReviewPallete.vue:120]: oversight

File: components/ReviewPallete.vue [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+  data () {
+    return {
+      editMode: false,
+      bufferRA: {
+        ques: '',
+        points: 0
+      },
+      rules: {
+        minLength: value => !!value || 'Required!',
+        weightageRule: (value) => {
```
Comment (king-11) [inline components/ReviewPallete.vue:138]: great job :))

File: pages/createassignment.vue [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+                  </v-card-title>
+                  <v-card-text>
+                    <v-container>
+                      <v-row>
+                        <div>
+                          <div>
+                            <v-btn outlined color="primary">
+                              <v-icon>mdi-cloud-upload-outline</v-icon>Add file
+                            </v-btn>
+                          </div>
```
Comment (king-11) [inline pages/createassignment.vue:105]: the teacher won't be using this

File: components/AddTask.vue [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+        }
+      }
+    }
+  },
+  computed: {
+    // for activation of some buttons.
+    validateTask () {
+      return this.task.title.length > 0 && this.task.allowedTypes.length > 0
+    },
+    // so that we can decide whether component is used for adding or editing the task.
```
Comment (king-11) [inline components/AddTask.vue:188]: nice use of computed for validation

**Review by king-11 — CHANGES_REQUESTED** [review]

File: components/AddTask.vue (hunk 4 above)
Comment (m-e-l-u-h-a-n) [reply components/AddTask.vue:7]: Title is diffrent `Edit Task` while editting the tak and `Add Task` while adding a new task that's why I have put it like this.

File: components/DateInput.vue (hunk 1 above)
Comment (m-e-l-u-h-a-n) [reply components/DateInput.vue:16]: Ohh.. I thought of doing it and then forgot :sweat_smile:

File: pages/createassignment.vue (hunk 8 above)
Comment (m-e-l-u-h-a-n) [reply pages/createassignment.vue:105]: Teacher would be able to see this as a preview of what task he/she has created like in google forms we can add a snackbar to inform that only student can submit it.
Or any other suggestion which could be better....

File: components/AddTask.vue (hunk 4 above)
Comment (king-11) [reply components/AddTask.vue:6]: no i meant v-layout's utility

File: pages/createassignment.vue (hunk 8 above)
Comment (king-11) [reply pages/createassignment.vue:109]: okay then will edit the component later when we add the API points just mark a comment somewhere for remembrance

**Comment by king-11** [comment]
LGTM :+1:  Finally :))

**Pull request merged**